packages/core/src/discovery.js

```diff
@@ -198,6 +198,16 @@ async function* captureSnapshotResources(page, snapshot, options) {
       }
     }
 
+    if (captureResponsiveDom) {
+      for (const device of captureResponsiveDom) {
+        yield waitForDiscoveryNetworkIdle(page, discovery);
```

Comment:
```suggestion
        // We are not adding these widths and pixels ratios in loop above because we want to explicitly reload the page after resize which we dont do above
        yield waitForDiscoveryNetworkIdle(page, discovery);
```